client/components/select-dropdown/style.scss

```diff
@@ -54,11 +54,30 @@
 			right: 13px;
 			top: 12px;
 
+		.is-compact & {
+			right: 4px;
+			top: 4px;
+		}
+
 		display: block;
 		line-height: 18px;
 		color: rgba( $gray, 0.5 );
 	}
 
+
+	.is-compact & {
+		padding: 7px;
+		color: darken( $gray, 10% );
+		font-size: 11px;
+		line-height: 1;
+		text-transform: uppercase;
+
+		.count {
```

Comment:
I think I'd remove the border, padding, and line-height from `count`:
![image](https://cloud.githubusercontent.com/assets/548849/11748045/9e9a7f44-a026-11e5-80b0-a1cdd335f6f3.png)

cc @rickybanister